docs/glossary.rst

```diff
@@ -0,0 +1,274 @@
+        are strongly specified, and behavior for a given set of inputs is deterministic
+        (within numerical stability). Outputs may not be well specified
+        until inputs are bound (*e.g.* until an instance is created).
+
+    graph state
+        The aggregation of state for operations (nodes) and data (edges),
+        constrained by directed
+        acyclic data flow topology. Granularity is not yet fully determined, but
+        state must account for completed and incomplete operations, and allow
+        distinction between idle and currently executing nodes.
+
+    operation
+    operation instance
+        A node in a work graph. Previously described as *WorkElement* or *node*.
+
+        Data sources and specific instances of operations are represented as
+        nodes in a work graph.
+
+        Identity: A node is the uniquely identifiable representation of a
+        :term:`function` instance, defined in terms of the inputs and
+        specified behavior.
+
+        Corollary: the definition of an operation (a node) is immutable once added to the graph.
+
+        Finer points:
+
+        * The outputs of an operation may be accessed and subscribed to at any time.
+        * Internally, operations may be stateful. They have metadata associated with
+          their degree of completion and, potentially, with references to other
+          resources to describe intermediate or final results.
+
+    port
+        Generic term for a named source, sink, resource, or binding hook on a node.
+
+    resource
+        Describes an API hook for an interaction mediated by a Context. Data flow
+        is described as *immutable* resources (generally produced as Operation outputs)
+        that can be consumed by binding to Operation inputs or by extracting as *results*
+        from the API. Some interactions cannot be represented in terms of producers
+        and subscribers of immutable data events: *Mutable* resources cannot be
+        managed by the Context as data events and require different work scheduling
+        policies that either (a) allows arbitrary (unscheduled) call-back through the API framework,
+        (b) dispatch the mutable resource collaboration to another Context, or (c)
+        allow operations to bind and interact with an interface not specified by the
+        API or not known to the responsible Context implementation. Examples include
+        the Context-provided *ensemble_reduce* functionality, the ensemble simulation
+        signaling facility (by which extension code can terminate a simulation early),
+        and the binding mechanism by which MD extension code can be attached to an
+        *MD* operation as a plugin. The nature of a resource is indicated by the
+        namespace of its *port* in the work record.
+
+    run time
+        .. todo:: Define the scope to be conveyed by the noun "run time".
+
+    simulation segment
+    trajectory segment
```

Comment:
@mrshirts writes:
> I'm not sure which is which. Is the simulation segment what happens between checkpoint intervals, or the overall unit? I don't think trajectory segment is a good synonym, since trajectory implies the PRODUCT of simulation, and simulation could describe either the process or the product.

It could be appropriate to split this into two terms...

The simulation segment is the shortest unit of simulator work that SCALE-MS concerns itself with, and it corresponds to the trajectory segment product of such work.